src/app/shell/personal-cabinet/provider/create-provider/create-provider.guard.ts

```diff
@@ -26,7 +26,7 @@ export class CreateProviderGuard implements CanDeactivate<unknown>, CanLoad {
     segments: UrlSegment[]): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
     const isEditMode = this.store.selectSnapshot<boolean>(AppState.isEditMode);
 
-    return isEditMode ? true : this.user$.pipe(map((user: User) => user.role === Role.provider && user.isRegistered === false));
+    return isEditMode ? true : this.user$.pipe(map((user: User) => user?.role === Role.provider && user.isRegistered === false));
```

Comment:
in which cases user can be undefined?

Reply:
thanks for catching that, I think it was for testing purposes, deleted '?'